main.go

```diff
@@ -0,0 +1,184 @@
+	})
+
+	block, _ := pem.Decode(rootCAs)
+	if block == nil {
+		log.Fatalf("CLIENT: Failed to parse certificate PEM")
+	}
+	CAuthority, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		log.Fatalf("CLIENT: Failed to parse certificate: %s", err.Error())
+	}
+
+	tlsConfigs := make(map[int]*tls.Config)
+
+	for i := 0; i < *concurrency; i++ {
+		_ , certPem := GenerateCertificate(CAuthority, parsedKey, privateKey, duration, fmt.Sprintf("localhost%d", i))
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(rootCAs)
+
+		cert, err := tls.X509KeyPair(certPem.Bytes(), privatePem.Bytes())
+		if err != nil {
+			log.Fatalf("CLIENT: Failed to load certificate and private key: %s", err.Error())
+		}
+
+		conf := &tls.Config{
+			Certificates: []tls.Certificate{cert},
+			ClientAuth: tls.RequireAnyClientCert,
+			ClientCAs: caCertPool,
+			InsecureSkipVerify: true,
+		}
+
+		conf.Rand = rand.Reader
+		tlsConfigs[i] = conf
+	}
+
+	for i := 0; i < *concurrency; i++ {
+		go ProcessOutgoingConnection(host, port, tlsConfigs[i], sleep)
+	}
+
+	fmt.Scanln()
+}
+
+func ProcessOutgoingConnection(host *string, port *int, conf *tls.Config, sleep *time.Duration)  {
+	Addr := fmt.Sprintf("%s%s%d", *host, ":", *port)
+
+	conn, err := tls.Dial("tcp", Addr, conf)
+
+	defer conn.Close()
+	if err != nil {
+		log.Fatalf("CLIENT: Failed to connect: %s", err.Error())
+	}
+
+	log.Printf("CLIENT: Connect to %s succeed", Addr)
+	time.Sleep(*sleep)
+
+	encoder := &Encoder{
+		JSONRPC: "2.0",
+		Method: "pki::RequestCertificate",
+		Params: map[string]string{
+			"ticket": " ",
+		},
+	}
+
+	message, err := json.Marshal(&encoder)
+
+	if err != nil{
+		log.Printf("CLIENT: JsonRpcConnection: Error while encoding JSRPC message. %s", err)
+	}
+
+	log.Printf("CLIENT: JsonRpcConnection: Sending pki::RequestCertificate JSRPC message: %s", message)
+
+	err = base.WriteNetStringToStream(conn, message)
+
+	if err != nil {
+		log.Printf("CLIENT: JsonRpcConnection: Error while sending json encoded JSRPC message: %s", err)
+	}
+}
+
+func GenerateCertificate(CACert *x509.Certificate, CAKey *rsa.PrivateKey, priv *rsa.PrivateKey, days *int, endpoint string) (*x509.Certificate, *bytes.Buffer) {
+	certificateTemplate := x509.Certificate{
+		SerialNumber:   big.NewInt(1658),
+		Subject: pkix.Name{
+			CommonName: endpoint,
+			Organization: []string{"Icinga GmbH"},
+			Country: []string{"DE"},
+			Province: []string{""},
+			Locality: []string{"Nuremberg"},
+
+		},
+		NotBefore:      time.Now(),
+		NotAfter:       time.Now().AddDate(0, 0, *days),
+		SubjectKeyId: []byte{1, 2, 3, 4, 6},
+		KeyUsage:       x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:    []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		IsCA:           false,
+		IPAddresses:    []net.IP{net.ParseIP("127.0.0.1"), net.IPv6loopback},
+	}
+
+	certBytes, err := x509.CreateCertificate(rand.Reader, &certificateTemplate, CACert, &priv.PublicKey, CAKey)
```

Comment:
Please dump a such certificate (probably in an extra program) and verify that this #1 task is done properly:

"Create (...) certificates (...) (**as Icinga 2 would do**)"